### PR TITLE
Set up Hono connection and digital twin as part of chart install

### PR DESCRIPTION
**Pull request by sophokles73**
The chart now uses a post-install hook to set up the AMQP 1.0 connection
to Hono and add a digitial twin for the demo-device to Ditto.

### CODE REVIEW

File: homepage/_packages/cloud2edge/tour.md [hunk 1]
```diff
@@ -19,38 +19,66 @@ or that you might experience some issues when testing.
...
+{% endclipboard %}
+
+## Retrieving the digital twin's current state
+
+The updated state of the digital twin can then be retrieved using:
+
+{% clipboard %}
+curl -u ditto:ditto http://$DITTO_API_IP:$DITTO_API_PORT_HTTP/api/2/things/org.eclipse.packages.c2e:demo-device
+{% endclipboard %}
+
```
Comment (thjaeckle) [inline homepage/_packages/cloud2edge/tour.md]: Additionally to the curl polling we could also add subscribing to the Ditto "SSE" (Server Sent Events) which gets each modification to the twin pushed:
```
{% clipboard %}
curl --http2 -u ditto:ditto -H "Accept:text/event-stream" -N http://$DITTO_API_IP:$DITTO_API_PORT_HTTP/api/2/things
{% endclipboard %}
```

Comment (sophokles73) [reply homepage/_packages/cloud2edge/tour.md]: That's even better, great idea :+1:

File: homepage/_packages/cloud2edge/scripts/setCloud2EdgeEnv.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+getService service-device-registry-ext "http https" REGISTRY
+getService adapter-amqp-vertx "amqp amqps" AMQP_ADAPTER
+getService adapter-coap-vertx "coap coaps" COAP_ADAPTER
+getService adapter-http-vertx "http https" HTTP_ADAPTER
+getService adapter-mqtt-vertx "mqtt secure-mqtt" MQTT_ADAPTER
+getService ditto-nginx "http" DITTO_API
+
+echo "# Run this command to populate environment variables"
+echo "# with the NodePorts of Hono's API endpoints:"
+echo "# eval \"\$(./nodePorts.sh RELEASE_NAME [NAMESPACE])\""
```
Comment (ctron) [inline homepage/_packages/cloud2edge/scripts/setCloud2EdgeEnv.sh]: I think the script is named `setCloud2EdgeEnv.sh`.

File: homepage/_packages/cloud2edge/scripts/setCloud2EdgeEnv.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+getService dispatch-router-ext "amqp amqps" AMQP_NETWORK
+getService service-device-registry-ext "http https" REGISTRY
+getService adapter-amqp-vertx "amqp amqps" AMQP_ADAPTER
+getService adapter-coap-vertx "coap coaps" COAP_ADAPTER
+getService adapter-http-vertx "http https" HTTP_ADAPTER
+getService adapter-mqtt-vertx "mqtt secure-mqtt" MQTT_ADAPTER
+getService ditto-nginx "http" DITTO_API
+
+echo "# Run this command to populate environment variables"
+echo "# with the NodePorts of Hono's API endpoints:"
```
Comment (thjaeckle) [inline homepage/_packages/cloud2edge/scripts/setCloud2EdgeEnv.sh]: It's not only about Hono's API endpoints, but API endpoints from the cloud2edge package.

**Comment by thjaeckle** [comment]
> As we already discussed, I would like to see the script running from inside a container on the cluster, so that people don't need to download the script. But that can be done in a future iteration.

As I understood the PR this is exactly what is done.
The script does not need to be executed by people themselves, but this is automatically done in `post-install-job.yaml` as Helm post install hook.

**Comment by sophokles73** [comment]
> The script does not need to be executed by people themselves, but this is automatically done in post-install-job.yaml as Helm post install hook.

I think @ctron is talking about the shell script for populating the environment vars with the endoint IPs/ports ...

**Pull request merged**